services/app-web/src/pages/SubmissionSummary/SubmissionSummary.tsx

```diff
@@ -302,11 +298,16 @@ export const SubmissionSummary = (): React.ReactElement => {
 
                 <ChangeHistory contract={contract} />
 
-                <UnlockSubmitModal
+                <ActionModal
                     modalRef={modalRef}
                     modalType="UNLOCK_CONTRACT"
                     submissionData={contract}
                 />
+                <ActionModal
```

Comment:
If you look at line 205, I have the approval modal there. The reason why it's up there is because I don't want to render the modal code if the user is not authorized. It's not required to do that, but it would be good to make it unavailable in the dom.

Reply:
We can use the `ActionModal` component, but we need to align the designs with what I have for my approval modal on line 205. My suggestion is to not have `ActionModal` handle all these functions.
- Placement of the modal should be toward the top of the page
- The modal submit button should be the default style.
- Modal width is longer than desgins.
- [Figma](https://www.figma.com/design/frKNnm6lkcpdjJ2eVJE8Bj/Managed-Care-Review?node-id=15083-79446&node-type=frame&t=jXUxzjSnklPJaQIV-0).

My opinion on the `ActionModal` is that it is doing too much. There was a reason for a common modal component to handle submit, unlock, and resubmit because they all shared the same design with different text. The APIs are closely related by updating the same status of the contract, but with approval, we are handling a different status of the contract, and the design is slightly different. With all this functionality put into one modal, it is increasing the complexity of the component and could be prone to bugs.

**Suggestion**: 
- Revert `ActionModal` back to `UnlockSubmitModal` or a better name. 
- Create a new modal component for the approval modal using the common `Modal` component.
- Place the modal above in the conditional rendering so it is not in the dom with unauthorized users.

Reply:
Yeah that makes sense to me, currently looking into breaking it out